### ISSUE TITLE
mayavi2 fails to run on nvidia drivers

### ISSUE THREAD

**Issue by OleksiiR** [enthought/mayavi#799]
Hello!

I cannot any visualization using mayavi2 because OpenGL fails to initialize when I use nvidia drivers. When I use nouveau drivers OpenGL initializes correctly in mayavi2.

Using nvidia drivers, I get the following output:

>>>LIBGL_DEBUG=verbose mayavi2 
********************************************************************************
WARNING: Imported VTK version (8.2) does not match the one used
         to build the TVTK classes (8.1). This may cause problems.
         Please rebuild TVTK.
********************************************************************************
                                                                                                                                                                                                                                             
libGL: screen 0 does not appear to be DRI2 capable                                                                                                                                                                                           
libGL: OpenDriver: trying /usr/lib64/dri/tls/swrast_dri.so                                                                                                                                                                                   
libGL: OpenDriver: trying /usr/lib64/dri/swrast_dri.so                                                                                                                                                                                       
libGL: Can't open configuration file /home/oleksii/.drirc: No such file or directory.                                                                                                                                                        
libGL: Can't open configuration file /home/oleksii/.drirc: No such file or directory.                                                                                                                                                        
libGL error: No matching fbConfigs or visuals found                                                                                                                                                                                          
libGL error: failed to load driver: swrast                                                                                                                                                                                                   
ERROR: In /home/abuild/rpmbuild/BUILD/VTK-8.2.0/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx, line 741                                                                                                                                        
vtkXOpenGLRenderWindow (0x55a145e392c0): GLEW could not be initialized: Missing GL version

I tried to substitute libGL in /usr/lib by the one in /usr/X11R6/lib, same with lib64. I tried to delete nouveau drivers completely as well. I tried to add paths to nvidia libs in LD_LIBRARY_PATH. None of these tricks helped. Please help.

OS: OpenSUSE Leap 15.0 64 bit (KDE)
Nvidia drivers are installed from Nvidia repository for OpenSUSE.

Best regards,
Oleksii

**Comment by prabhuramachandran** [comment]
This seems like a driver issue, I suspect you will have the same problem if you run glxgears.  This is not a Mayavi related issue so unfortunately, there isn't much we can do about this.

**Comment by OleksiiR** [comment]
Thanks for your reply. The problem is that glxgears is running with no problems, same as glxspheres and even ParaView or Salome. For instance, here is glxgears output (I have cut the piece with GL_EXTENSIONS):

>>> glxgears -info
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
GL_RENDERER   = GeForce 210/PCIe/SSE2
GL_VERSION    = 3.3.0 NVIDIA 340.107
GL_VENDOR     = NVIDIA Corporation
GL_EXTENSIONS = ... 
VisualID 39, 0x27

So, in general, OpenGL is running OK, this seems to be mayavi2 specific issue.

**Comment by prabhuramachandran** [comment]
How did you install VTK?  Also that initial warning is suggesting that you somehow have two VTK versions, you seem to have VTK 8.2 and have installed Mayavi with the older VTK 8.1.  I suspect you have a possibly broken VTK installation and need more details to find out.

**Comment by prabhuramachandran** [comment]
I suggest uninstalling the VTK-8.2 version and using the vtk wheels from pypi (pip install vtk).  That should work -- it does for me on my linux machine with an nVidia card.

**Comment by OleksiiR** [comment]
Thanks for suggestion. Previously I installed everything from OpenSuse repositories. Now I deleted vtk and ran "sudo pip install vtk". It went fine. But I could not install mayavi by "sudo pip install mayavi". I got: 

...
  running gen_docs
  QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
  qt.qpa.screen: QXcbConnection: Could not connect to display
  Could not connect to any X display.
  
  ----------------------------------------
  Failed building wheel for mayavi
  Running setup.py clean for mayavi
Failed to build mayavi
Installing collected packages: mayavi
  Running setup.py install for mayavi ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-9iwmo942/mayavi/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-srwe7v8s/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
...
running gen_docs
    QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
    qt.qpa.screen: QXcbConnection: Could not connect to display
    Could not connect to any X display.
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-9iwmo942/mayavi/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-srwe7v8s/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-9iwmo942/mayavi/
